aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -185,15 +186,12 @@ private boolean isUnauthorizedException(DataStoreException exception) {
                 },
                 emitter::onNext,
                 dataStoreException -> {
-                    // Only call onError if the Observable hasn't been disposed and it's not an Unauthorized error.
-                    // Unauthorized errors are ignored, so that DataStore can still be used even if the user is only
+                    // Ignore Unauthorized errors, so that DataStore can still be used even if the user is only
                     // authorized to read a subset of the models.
-                    if (!emitter.isDisposed()) {
-                        if (isUnauthorizedException(dataStoreException)) {
-                            LOG.warn("Unauthorized failure for " + subscriptionType + " " + modelSchema.getName());
-                        } else {
-                            emitter.onError(dataStoreException);
-                        }
+                    if (isUnauthorizedException(dataStoreException)) {
+                        LOG.warn("Unauthorized failure for " + subscriptionType.name() + " " + modelSchema.getName());
+                    } else {
+                        onFailure.accept(dataStoreException);
```

Comment:
This is the most important change in this whole PR.  Instead of passing errors to the `emitter`, and then to the `buffer`, all errors are sent directly to an `onFailure` callback which the Orchestrator passes in to the `SubscriptionProcessor` on initialization.  

Basically, we are no longer sending errors to any stream, since those streams end up getting disposed after receiving an error.